include/deal.II/base/patterns.h

```diff
@@ -1858,6 +1916,99 @@ namespace Patterns
         return *m.begin();
       }
     };
+
+    // Tuples
+    template <class... Args>
+    struct Convert<std::tuple<Args...>>
+    {
+      typedef std::tuple<Args...> T;
+
+      static std::unique_ptr<Patterns::PatternBase> to_pattern()
+      {
+        static_assert(internal::RankInfo<T>::map_rank > 0,
+                      "Cannot use this class for non tuple-compatible types.");
+        return std_cxx14::make_unique<Patterns::Tuple>(
+                 internal::default_map_separator[internal::RankInfo<T>::map_rank-1],
+                 *Convert<Args>::to_pattern()...);
+      }
+
+      template<std::size_t... I>
+      static
+      std::array<std::string, std::tuple_size<T>::value>
+      to_string_internal_1(const T &t,
```

Comment:
could you make this function and the next `private`? they seem like internal details.